apmpackage/apm/data_stream/app_metrics/elasticsearch/ingest_pipeline/default.yml

```
@@ -23,6 +23,8 @@ processors:
             dynamic_templates[name] = "histogram";
           } else if (metric_type == "summary") {
             dynamic_templates[name] = "summary";
+          } else if (metric_type == "counter" || metric_type == "gauge") {
+            dynamic_templates[name] = "doubles";
```

```suggestion
            dynamic_templates[name] = "double";
```

Singular, for consistency with "summary" and "histogram".